### PR TITLE
Implement DPT 13.xxx types

### PR DESCRIPTION
**Pull request by DocLambda**
Implement DPT 13.xxx documented in KNX standard v2.1.

### CODE REVIEW

File: knx/dpt/formats.go [hunk 1]
```diff
@@ -108,3 +108,25 @@ func unpackU32(data []byte, i *uint32) error {
...
 }
+
+func packV32(i int32) []byte {
+	b := make([]byte, 5)
+
+	b[0] = 0
+	b[1] = byte((i >> 24) & 0xff)
+	b[2] = byte((i >> 16) & 0xff)
+	b[3] = byte((i >> 8) & 0xff)
+	b[4] = byte(i & 0xff)
```
Comment (vapourismo) [inline knx/dpt/formats.go]: Have you tested this for corner cases like `-1` and `MinInt32`?

File: knx/dpt/types.go [hunk 2]
```diff
@@ -316,3 +316,211 @@ func (d DPT_12001) Unit() string {
...
+func (d *DPT_13001) Unpack(data []byte) error {
+	var value int32
+
+	if err := unpackV32(data, &value); err != nil {
+		return err
+	}
+	*d = DPT_13001(value)
+
+	return nil
+}
```
Comment (vapourismo) [inline knx/dpt/types.go]: `return unpackV32(data, (*int32)(d))` would be much clearer.

File: knx/dpt/types.go [hunk 3]
```diff
@@ -316,3 +316,211 @@ func (d DPT_12001) Unit() string {
...
+func (d *DPT_13002) Unpack(data []byte) error {
+	var value int32
+
+	if err := unpackV32(data, &value); err != nil {
+		return err
+	}
+	*d = DPT_13002(value)
+
+	return nil
+}
```
Comment (vapourismo) [inline knx/dpt/types.go]: This can be simplified analog to `func (d *DPT_13001) Unpack(data []byte) error`.

File: knx/dpt/types.go [hunk 4]
```diff
@@ -316,3 +316,211 @@ func (d DPT_12001) Unit() string {
...
+func (d *DPT_13011) Unpack(data []byte) error {
+	var value int32
+
+	if err := unpackV32(data, &value); err != nil {
+		return err
+	}
+	*d = DPT_13011(value)
+
+	return nil
+}
```
Comment (vapourismo) [inline knx/dpt/types.go]: Simplify as well, please.

File: knx/dpt/formats.go (hunk 1 above)
Comment (DocLambda) [reply knx/dpt/formats.go]: Will check. Btw. Why do we get 5 byte here (same for float type DPT9.xxx) with the first byte always zero (as far as I observed) while the datatype is just 4 byte?

**Comment by DocLambda** [comment]
Thanks for your comments! Will fix the code according to your suggestions. However I'm afk for the whole week, so don't expect an update before the weekend...

File: knx/dpt/formats.go (hunk 1 above)
Comment (vapourismo) [reply knx/dpt/formats.go]: Because the KNX/net protocol is a leaky abstraction. The 2 most significant bits of the first byte are reserved. They encode the two least significant bits of the ACPI command (group read, group write, etc) once the entire frame is assembled.

In the case of V32 for instance, you actually have 38bits for data.

The reason this is also present in this package is that it gives us a way to differentiate between 6bits of data (1byte) and >6bits (>1byte).

Comment (DocLambda) [reply knx/dpt/formats.go]: Added testing of corner-cases (MinInt32, MaxInt32) to unit-tests. Furthermore, all unit-test for DPT13.xxx test for both positive and negative random samples.
However, I cannot test with negative real-world data as I only do own sensors that submit positive DPT13.xxx values.

File: knx/dpt/types.go (hunk 2 above)
Comment (DocLambda) [reply knx/dpt/types.go]: Done
